src/HDF5.jl

```diff
@@ -1459,73 +1465,70 @@ function getindex(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, r::HDF5Refere
     h5object(obj_id, parent)
 end
 
-# Helper for reading compound types
-function read_row(io::IO, membertype, membersize)
-    row = Any[]
-    for (dtype, dsize) in zip(membertype, membersize)
-        if dtype === String
-            push!(row, unpad(read!(io, Vector{UInt8}(undef,dsize)), H5T_STR_NULLPAD))
-        elseif dtype<:HDF5.FixedArray && eltype(dtype)<:HDF5BitsKind
-            val = read!(io, Vector{eltype(dtype)}(undef,prod(size(dtype))))
-            push!(row, reshape(val, size(dtype)))
-        elseif dtype<:HDF5BitsKind
-            push!(row, read(io, dtype))
-        else
-            # for other types, just store the raw bytes and let the user
-            # decide what to do
-            push!(row, read!(io, Vector{UInt8}(undef,dsize)))
-        end
+# convert Cstring/FixedString to String
+function normalize_types(x::NamedTuple{T}) where T
```

Comment:
where was this normalization obtained from?

Reply:
this was taken from the current handling of strings/vlen types.

Reply:
except that for vlen arrays I use a non-owning unsafe wrap and then make a copy. I think this is the right thing to do since hdf5 wants to manage its own memory.